src/main/resources/category/java/common.xml

```diff
@@ -5,35 +5,26 @@
     <rule name="AvoidCDIReferenceLeak"
           language="java"
           message="Explicit CDI references need to be destroyed otherwise they leak."
-          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
           typeResolution="true"
           externalInfoUrl="${doc_root}/JavaCodePerformance.md#pml05">
         <description>Problem: A proxy object is created by Contexts and Dependency Injection (CDI) for explicit references, they are not de-referenced implicitly and become a memory leak. &#13;
             Solution: Destroy the reference explicitly.
         </description>
         <priority>1</priority>
         <properties>
-            <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
-            <property name="tag" value="performance" type="String" description="for-sonar"/>
-            <!--property name="version" value="2.0"/-->
+            <property name="pmd-version" value="7" type="String" description="for-compatibility"/>
+            <property name="tag1" value="jpinpoint-rule" type="String" description="for-sonar"/>
+            <property name="tag2" value="performance" type="String" description="for-sonar"/>
             <property name="xpath">
                 <value><![CDATA[
-//Expression/PrimaryExpression/PrimaryPrefix/Name[@Image='CDI.current']/../../
-PrimarySuffix[@Image = 'select'
+//MethodCall[@MethodName='select']/MethodCall[@MethodName='current']/TypeExpression/ClassType[@SimpleName='CDI'
 and not
-(ancestor::MethodDeclaration//TryStatement/FinallyStatement//PrimaryExpression[
-PrimaryPrefix/Name/@Image='CDI.current'][PrimarySuffix/@Image = 'destroy']
-[PrimarySuffix/Arguments/ArgumentList/Expression/PrimaryExpression/PrimaryPrefix/Name/@Image =
-ancestor::MethodDeclaration//Expression/PrimaryExpression/PrimaryPrefix/Name[@Image='CDI.current']/../../
-PrimarySuffix[@Image = 'select']/ancestor::StatementExpression/PrimaryExpression/PrimaryPrefix/Name/@Image
-])
-and not
-(ancestor::MethodDeclaration//TryStatement/FinallyStatement//PrimaryExpression[
-PrimaryPrefix/Name/@Image='CDI.current'][PrimarySuffix/@Image = 'destroy']
-[PrimarySuffix/Arguments/ArgumentList/Expression/PrimaryExpression/PrimaryPrefix/Name/@Image =
-ancestor::MethodDeclaration//VariableInitializer/Expression/PrimaryExpression/PrimaryPrefix/Name[@Image='CDI.current']/../../
-PrimarySuffix[@Image = 'select']/ancestor::VariableDeclarator/VariableDeclaratorId/@Name
-])]
+(ancestor::MethodDeclaration//TryStatement/FinallyClause
+    //MethodCall[@MethodName='destroy'][ArgumentList/VariableAccess[@Name=ancestor::
+        MethodDeclaration//(VariableDeclarator|AssignmentExpression)[.//MethodCall[@MethodName='current']/TypeExpression/ClassType[@SimpleName='CDI']]/(VariableId|VariableAccess)/@Name]]
+    [MethodCall[@MethodName='current']/TypeExpression/ClassType[@SimpleName='CDI']])
+]
```

Comment:
Seems more concise and shorter for pmd7... useful it seems: `MethodCall[@MethodName]` etc...
Good to have unit tests to migrate :-)

Reply:
yes, indeed :-)